openassessment/xblock/static/js/src/studio/oa_edit.js

```diff
@@ -268,19 +272,45 @@ export class StudioView {
   }
 
   /**
-     Mark validation errors.
+     Perform validation on each view and determine views failing validation.
 
      Returns:
-     Boolean indicating whether the view is valid.
+     List of tabs failing validation or empty list
 
      * */
   validate() {
-    const settingsValid = this.settingsView.validate();
-    const assessmentsStepsValid = this.assessmentsStepsView.validate();
-    const scheduleValid = this.scheduleView.validate();
-    const rubricValid = this.rubricView.validate();
-    const promptsValid = this.promptsView.validate();
-    return settingsValid && assessmentsStepsValid && scheduleValid && rubricValid && promptsValid;
+    const tabsToValidate = [
+      this.promptsView,
+      this.rubricView,
+      this.scheduleView,
+      this.assessmentsStepsView,
+      this.settingsView,
+    ];
+    const tabsFailingValidation = [];
+
+    tabsToValidate.forEach((tab) => {
+      if (!tab.validate()) {
+        tabsFailingValidation.push(tab);
+      }
+    });
+
+    return tabsFailingValidation;
+  }
+
+  /** Given a list of tabs failing validation, mark a tab as invalid */
+  markTabsWithValidationErrors(tabsFailingValidation) {
+    tabsFailingValidation.forEach((tab) => {
+      this.markTabAsInvalid(tab.tabTarget);
+    });
+  }
+
+  /**
+   * Given an aria controls target, add or remove validation warning styling.
+   *  value - true enables invalid styling, false removes invalid styling
+   * */
+  markTabAsInvalid(controlTarget, value = true) {
+    const tab = $(`[aria-controls='${controlTarget}']`, this.element);
```

Comment:
I like that the `href` approach is more semantically accurate but it is also more fragile since it becomes inherently more tied to the DOM structure. I'm thinking IDs are the way to go but, candidly, all of this is a cluster...